n_helpers.c

```diff
@@ -84,205 +84,89 @@ NOTE_C_STATIC void setTime(JTIME seconds);
 
-    // Examine "cobs" from the response to evaluate the space required to hold
-    // the COBS-encoded data to be received from the Notecard.
-    long int cobs = JGetInt(rsp, "cobs");
-    JDelete(rsp);
-    if (!cobs) {
-        // If cobs is 0, the required buffer length is 0 because there's nothing
-        // to receive.
-        *len = 0;
+    // Validate parameter(s)
+    if (encData == NULL || decBuf == NULL) {
+        NOTE_C_LOG_ERROR(ERRSTR("NULL parameter", c_err));
+        result = 0;
+    } else if (decBufSize < cobsGuaranteedFit(encDataLen)) {
+        NOTE_C_LOG_ERROR(ERRSTR("output buffer too small", c_err));
+        result = 0;
     } else {
-        // Otherwise, the required length is cobs + 1: the binary data plus
-        // 1 byte for the terminating newline.
-        *len = cobs + 1;
+        result = cobsDecode((uint8_t *)encData, encDataLen, NOTE_C_BINARY_EOP, decBuf);
     }
 
-    return NULL;
+    return result;
 }
 
 //**************************************************************************/
 /*!
-  @brief  Reset the Notecard's binary buffer.
-
-  @returns  NULL on success, else an error string pointer.
-
-  @note  This operation is necessary to clear the Notecard's binary buffer after
-         a binary object is received from the Notecard, or if the Notecard's
-         binary buffer has been left in an unknown state due to an error arising
-         from a binary transfer to the Notecard.
- */
-/**************************************************************************/
-const char * NoteBinaryDataReset(void)
-{
-    J *req = NoteNewRequest("card.binary");
-    if (req) {
-        JAddBoolToObject(req, "delete", true);
 
-        // Ensure the transaction doesn't return an error.
-        J *rsp = NoteRequestResponse(req);
-        if (NoteResponseError(rsp)) {
-            NOTE_C_LOG_ERROR(JGetString(rsp,"err"));
-            JDelete(rsp);
-            NOTE_C_LOG_ERROR("failed to reset binary buffer");
-            return ERRSTR("failed to reset binary buffer", c_err);
-        }
-    } else {
-        NOTE_C_LOG_ERROR("unable to allocate request");
-        return ERRSTR("unable to allocate request", c_mem);
-    }
+  @brief  Encode binary data to transmit to the Notecard.
 
-    return NULL;
-}
-
-//**************************************************************************/
-/*!
-  @brief  Decode a binary payload received from the Notecard.
-
-  @param  inBuf The binary payload.
-  @param  inLen The length of the binary payload.
-  @param  outBuf The buffer to write the decoded payload to. This can be the
-                 same address as inBuf, allowing for in-place decoding.
-  @param  outLen On input, holds the length of outBuf. On output, holds the
-                 length of the decoded data.
+  @param  decData The decoded binary data to encode.
+  @param  decDataLen The length of the decoded binary data.
+  @param  encBuf The target buffer for the encoded data. This can be in the same
```

Comment:
This can be in the same - > This can be the same

Reply:
I deliberately chose to add "in". The address MUST be different, but the addresses can be part of (or "in") the same buffer.

Reply:
I reworded it to be more clear and explicit regarding the concern expressed above.